provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/issue/BlackDuckIssueTrackerCallbackHandler.java

```diff
@@ -0,0 +1,108 @@
+        Optional<BlackDuckApiClient> optionalBlackDuckApiClient = createBlackDuckProperties(callbackInfo.getProviderConfigId())
+                                                                      .flatMap(this::createBlackDuckServicesFactory)
+                                                                      .map(BlackDuckServicesFactory::getBlackDuckApiClient);
+        if (optionalBlackDuckApiClient.isPresent()) {
+            BlackDuckProviderIssueHandler blackDuckProviderIssueHandler = new BlackDuckProviderIssueHandler(gson, optionalBlackDuckApiClient.get());
+            BlackDuckProviderIssueModel issueModel = createBlackDuckIssueModel(event);
+            createOrUpdateBlackDuckIssue(blackDuckProviderIssueHandler, issueModel, callbackInfo);
+        }
+        logger.debug("Finished handling issue-tracker callback event with id '{}'", eventId);
+    }
+
+    private Optional<BlackDuckProperties> createBlackDuckProperties(Long providerConfigId) {
+        return configurationAccessor.getConfigurationById(providerConfigId)
+                   .map(blackDuckPropertiesFactory::createProperties);
+    }
+
+    private Optional<BlackDuckServicesFactory> createBlackDuckServicesFactory(BlackDuckProperties blackDuckProperties) {
+        IntLogger intLogger = new Slf4jIntLogger(logger);
+        try {
+            BlackDuckHttpClient blackDuckHttpClient = blackDuckProperties.createBlackDuckHttpClient(intLogger);
+            BlackDuckServicesFactory blackDuckServicesFactory = blackDuckProperties.createBlackDuckServicesFactory(blackDuckHttpClient, intLogger);
+            return Optional.of(blackDuckServicesFactory);
+        } catch (AlertException e) {
+            logger.error("Failed to create a BlackDuck http client", e);
+            return Optional.empty();
+        }
+    }
+
+    private void createOrUpdateBlackDuckIssue(BlackDuckProviderIssueHandler blackDuckProviderIssueHandler, BlackDuckProviderIssueModel issueModel, IssueTrackerCallbackInfo callbackInfo) {
+        try {
+            blackDuckProviderIssueHandler.createOrUpdateBlackDuckIssue(issueModel, callbackInfo.getCallbackUrl(), callbackInfo.getBlackDuckProjectVersionUrl());
+        } catch (IntegrationException e) {
+            logger.error("Failed to create or update BlackDuck issue: {}", issueModel, e);
+        }
+    }
+
+    private BlackDuckProviderIssueModel createBlackDuckIssueModel(IssueTrackerCallbackEvent event) {
+        String blackDuckIssueStatus = mapOperationToAlertStatus(event.getOperation());
+        return new BlackDuckProviderIssueModel(event.getIssueKey(), blackDuckIssueStatus, event.getIssueSummary(), event.getIssueUrl());
+    }
+
+    private String mapOperationToAlertStatus(IssueOperation issueOperation) {
+        switch (issueOperation) {
+            case OPEN:
+            case UPDATE:
+                return "Created by Alert";
+            case RESOLVE:
+                return "Resolved by Alert";
```

Comment:
We may benefit by adding the string associated here to the IssueOperation class. That way we can just call the specific method from the Enum instead of a switch

Reply:
I went back and forth on this, but I think this logic does not belong in `IssueOperation` because it is only related to the callback handling.